### PR TITLE
Allow " in password

### PR DESCRIPTION
**Pull request by TheREK3R**

### CODE REVIEW

**Comment by ramereth** [comment]
@TheREK3R you really should split this into multiple commits. I.e. one that makes all the suites run and then one that adds the actual fix for the issue.

**Comment by ramereth** [comment]
@TheREK3R I'm still confused with the context of the changes you made. Are you able to split them out a little more and be more descriptive in each commit for the change?

**Comment by TheREK3R** [comment]
I added more descriptive commit messages, but I am not sure what is confusing? In the first commit they mangled the service name, it is supposed to be mariadb but they created a conflict by making the mysql service. In the second the error is that the " is not escaped '\"' so it is interpreted as end of command by sql. Other symbols will give this same error though so I dont know if there is a ruby function for that or if I should creatre a helper?

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -13,32 +13,32 @@ provisioner:
   product_version: <%= ENV['CHEF_VERSION'] || 'latest' %>
   install_strategy: once
   enforce_idempotency: true
-  multiple_converge: 2
+  multiple_converge: 3
```
Comment (ramereth) [inline kitchen.dokken.yml]: Why was this changed?

File: kitchen.yml [hunk 2]
```diff
@@ -4,9 +4,9 @@ driver:
 
 provisioner:
   name: chef_solo
-  product_name: chef
+  # product_name: chef
```
Comment (ramereth) [inline kitchen.yml]: Why was this changed?

File: resources/server_install.rb [hunk 3]
```diff
@@ -51,7 +51,7 @@
   #                                             2- the user did not pass anything to the password argument OR
   #                                                the user did not define node['mariadb']['server_root_password'] attribute
   mariadb_root_password = (new_resource.password == 'generate' || new_resource.password.nil?) ? secure_random : new_resource.password
-
+  mariadb_root_password = mariadb_root_password.gsub('\\', '\\\\\\').gsub('\'', '\\\\\'').gsub('"', '\\\\"')
```
Comment (ramereth) [inline resources/server_install.rb:54]: I feel like this could be done better but I'm not sure how.

File: resources/server_install.rb [hunk 4]
```diff
@@ -76,7 +76,6 @@
     group 'mysql'
     mode '755'
     recursive true
-    action :nothing
```
Comment (ramereth) [inline resources/server_install.rb]: Why was this removed?

File: kitchen.dokken.yml [hunk 5]
```diff
@@ -16,22 +16,19 @@ provisioner:
...
-    driver:
-      image: dokken/amazonlinux-2
-      pid_one_command: /usr/lib/systemd/systemd
-
   - name: centos-7
     driver:
       image: dokken/centos-7
       platform: rhel
       pid_one_command: /usr/lib/systemd/systemd
+    provisioner:
```
Comment (ramereth) [inline kitchen.dokken.yml]: Please remove this and the one below.

File: kitchen.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ provisioner:
   name: chef_solo
   product_name: chef
   enforce_idempotency: true
-  multiple_converge: 2
+  multiple_converge: 3
```
Comment (ramereth) [inline kitchen.yml]: What specifically is causing us to need to bump this to 3 globally?

**Review by ramereth — CHANGES_REQUESTED** [review]

File: kitchen.yml (hunk 6 above)
Comment (TheREK3R) [reply kitchen.yml]: oh sorry, i missed that one, thought it was only in the dokken yml. I can test again but what it was is that the debian/ubuntu ones wouldnt be idempotent until the third converge

Comment (TheREK3R) [reply kitchen.yml]: I must have fixed something after that that allowed them to though so then its vestigial now

Comment (ramereth) [reply kitchen.yml]: Then I recommend you revert this back to 2.

File: kitchen.dokken.yml [hunk 7]
```diff
@@ -16,11 +16,6 @@ provisioner:
   multiple_converge: 2
 
 platforms:
-  - name: amazonlinux-2
-    driver:
-      image: dokken/amazonlinux-2
-      pid_one_command: /usr/lib/systemd/systemd
-
```
Comment (ramereth) [inline kitchen.dokken.yml:23]: Why is this being removed?

Comment (TheREK3R) [reply kitchen.dokken.yml:23]: It isnt listed as supported in the metadata so I removed it

**Comment by ramereth** [comment]
@TheREK3R go ahead and please make an upstream PR
